controllers/apps/operations/datascript.go

```diff
@@ -0,0 +1,283 @@
+				return appsv1alpha1.OpsSucceedPhase
+			} else if condition.Type == batchv1.JobFailed && condition.Status == corev1.ConditionTrue {
+				return appsv1alpha1.OpsFailedPhase
+			}
+		}
+		return appsv1alpha1.OpsRunningPhase
+	}
+
+	// retrieve job for this opsRequest
+	jobList := &batchv1.JobList{}
+	err := cli.List(reqCtx.Ctx, jobList, client.InNamespace(cluster.Namespace), client.MatchingLabels(getDataScriptJobLabels(cluster.Name, spec.ComponentName, opsRequest.Name)))
+	if err != nil {
+		return appsv1alpha1.OpsFailedPhase, 0, err
+	}
+
+	if len(jobList.Items) == 0 {
+		return appsv1alpha1.OpsFailedPhase, 0, fmt.Errorf("job not found")
+	}
+	// check job status
+	job := &jobList.Items[0]
+	phase := getStatusFromJobCondition(job)
+	// jobs are owned by opsRequest, so we don't need to delete them explicitly
+	if phase == appsv1alpha1.OpsFailedPhase {
+		return phase, 0, fmt.Errorf("job execution failed, please check the job log with `kubectl logs jobs/%s -n %s`", job.Name, job.Namespace)
+	} else if phase == appsv1alpha1.OpsSucceedPhase {
+		return phase, 0, nil
+	}
+	return phase, time.Second, nil
+}
+
+func (o DataScriptOpsHandler) ActionStartedCondition(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) (*metav1.Condition, error) {
+	return appsv1alpha1.NewDataScriptCondition(opsRes.OpsRequest), nil
+}
+
+func (o DataScriptOpsHandler) SaveLastConfiguration(reqCtx intctrlutil.RequestCtx, cli client.Client, opsResource *OpsResource) error {
+	return nil
+}
+
+func (o DataScriptOpsHandler) GetRealAffectedComponentMap(opsRequest *appsv1alpha1.OpsRequest) realAffectedComponentMap {
+	return realAffectedComponentMap(opsRequest.Spec.GetDataScriptComponentNameSet())
+}
+
+// getScriptContent will get script content from script or scriptFrom
+func getScriptContent(reqCtx intctrlutil.RequestCtx, cli client.Client, spec *appsv1alpha1.ScriptSpec) ([]string, error) {
+	script := make([]string, 0)
+	if len(spec.Script) > 0 {
+		script = append(script, spec.Script...)
+	}
+	if spec.ScriptFrom == nil {
+		return script, nil
+	}
+	configMapsRefs := spec.ScriptFrom.ConfigMapRef
+	secretRefs := spec.ScriptFrom.SecretRef
+
+	if len(configMapsRefs) > 0 {
+		obj := &corev1.ConfigMap{}
+		for _, cm := range configMapsRefs {
+			if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: reqCtx.Req.Namespace, Name: cm.Name}, obj); err != nil {
+				return nil, err
+			}
+			script = append(script, obj.Data[cm.Key])
+		}
+	}
+
+	if len(secretRefs) > 0 {
+		obj := &corev1.Secret{}
+		for _, secret := range secretRefs {
+			if err := cli.Get(reqCtx.Ctx, types.NamespacedName{Namespace: reqCtx.Req.Namespace, Name: secret.Name}, obj); err != nil {
+				return nil, err
+			}
+			if obj.Data[secret.Key] == nil {
+				return nil, fmt.Errorf("secret %s/%s does not have key %s", reqCtx.Req.Namespace, secret.Name, secret.Key)
+			}
+			secretData := string(obj.Data[secret.Key])
+			// trim the last \n
```

Comment:
do we need to decode the secret data?

Reply:
the data retrieved as string is decoded.